### DISCUSSION TITLE
How does the desktop app work?

### DISCUSSION THREAD

**Discussion by sgarner** [warrenday/graphql-network-inspector#72] [Q&A] [answered]
Thanks for making this app. I've got the Chrome extension working, and wanted to try the desktop app. But no requests are appearing and I don't understand how it's supposed to work (the docs are a little light on this point).

I get the impression that it will intercept all network traffic on my machine and scan for GraphQL queries. That sounds... very scary. Wouldn't it be better if the proxy just listened on a particular port and I could choose what traffic to send to it?

In any case, how can I troubleshoot to figure out why no requests are detected in the desktop app, even though they are detected in the Chrome extension?

**Comment by warrenday** [comment]
Hey @sgarner the desktop app works on the same principal as all other proxy tools like Charles Proxy. The only difference being we filter and display GraphQL traffic only. This happens all locally on your machine and data is never sent anywhere else, so it remains secure. The main benefit being that with desktop you can also intercept and modify in-flight requests.

If you're proxying SSL traffic such as from a website like https://medium.com/ you'll need to enable SSL using the wizard in the top right corner of the app.

<img width="864" alt="Screenshot 2022-06-14 at 08 09 15" src="https://user-images.githubusercontent.com/2871038/173515495-b5db4b23-066d-49db-bf11-640e21b8fb1c.png">
<img width="857" alt="Screenshot 2022-06-14 at 08 09 47" src="https://user-images.githubusercontent.com/2871038/173515566-7c9a3df7-dd33-4a7f-a1de-fab08ae5b1ec.png">

If you're using localhost mac does not proxy the traffic, so you need to add an entry to `/private/etc/hosts` to use a different url.
<img width="678" alt="Screenshot 2022-06-14 at 08 12 28" src="https://user-images.githubusercontent.com/2871038/173516338-3106b180-5c49-4395-a038-c6e676a1be65.png">

With this for example instead of visiting `http://localhost:3000` I would visit `http://graphproxy.local:3000` (the network traffic must also go to this URL)

If you're having trouble I'd be happy to jump on a google meet with you.

Thanks

Warren

**Comment by sgarner** [comment, marked as answer]
> the desktop app works on the same principal as all other proxy tools like Charles Proxy

Ahh, okay, I get it now.

So the configuration step that I was missing was to set the `(Secure) Web Proxy` to `127.0.0.1:8000` in macOS System Preferences > Network.

<img width="644" alt="image" src="https://user-images.githubusercontent.com/904004/173710585-e98d3f83-0c7d-4ef4-8ab0-13c12241f15f.png">

It appears the installer did this automatically for my Wi-Fi connection, but not for my wired Ethernet connection. Since my Ethernet connection has higher priority, my internet traffic wasn't using the proxy until I copied the settings across. Now it is working.

> This happens all locally on your machine and data is never sent anywhere else, so it remains secure.

Good to know. Thanks!

**Reply by sgarner** [reply]
> It appears the installer did this automatically

Correction: I notice the app enables and disables the proxy settings for Wi-Fi when the app is opened and closed. Without that functionality for Ethernet, I have to manually disable the proxy after closing the app or my internet stops working. So this should probably be considered a bug...

**Reply by warrenday** [reply]
Ahh yes definitely a bug. All of the machines tested are on wifi so Ethernet was clearly missed. Thanks for the details, it really helps.

I'll push an update shortly.

Glad to hear you have it working in the meantime though.

**Comment by warrenday** [comment]
Hey @sgarner i've pushed a fix to auto configure other network types. This should mean you no longer need to manually configure the proxy for ethernet or any other connection. You can grab the latest version here.
https://github.com/warrenday/graphql-network-inspector-desktop/releases/tag/v0.4.6
